scipy/stats/contingency.py

```diff
@@ -287,7 +287,10 @@ def chi2_contingency(observed, correction=True, lambda_=None):
     else:
         if dof == 1 and correction:
             # Adjust `observed` according to Yates' correction for continuity.
-            observed = observed + 0.5 * np.sign(expected - observed)
+            # Implementation is compliant with R's one
+            # https://github.com/wch/r-source/blob/trunk/src/library/stats/R/chisq.test.R
```

Comment:
I am afraid that by saying this, you are actually preventing us to integrate this change due to the GPL licence of `r-source`.

cc @rgommers. Can we do something here?

Reply:
Indeed. We can test again R, but cannot look at its source code or integrate Python code that you wrote based on the GPL'd R code. Not much we can do about this I'm afraid.

Reply:
In such cases, should we close the PR and propose to re-open a new one?

Reply:
I can prepare the same pull request without mentioning R's implementation (i.e. be silent) or  implement the original correction, e.g. as follows:

```
            # Adjust `observed` according to Yates' correction for continuity.
            yates = 0.5 * np.minimum(1.0, np.abs(expected - observed))
            observed = observed + yates * np.sign(expected - observed)
```

The current SciPy's implementation of Yates' correction shifts observed values toward expected by 0.5 (disregard of difference between observed and expected). Sometimes (see case mentioned in gh-#13875)  these shifts became very large and lead to unreliable results.  R's implementation uses `min(0.5, abs(observed-expected))`. This perfectly handles such cases. We  can yet propose another one, e.g. shift by 0.5 or 0.5 * |observed - expected|, what is lesser. In this case we will get results which differ from R's one (in some cases), but desired behavior will be met: p-values obtained with correction became greater than p-values obtained without it. (in tests, we wouldn't refer to R's implementation results, but only check `p-val (with correction) > p-val (without correction)` and `chi2 (with correction) < chi2 (without correction)`. 

In case of X:
`X = array([[1000,    1],
       [  10,    0]])`

`chi2_contingency(X, correction=False)` :

```
chi2 = 0.009999901089009998  # current SciPy's implementation
p-value = 0.9203447180766172  # current SciPy's implementation
```

`chi2_contingency(X, correction=True)` :
```
chi2 = 24.551782153490077   # current SciPy's implementation
p-value = 7.233940557522228e-07  # current SciPy's implementation
```

R's  `chisq.test` with Yates' correction:

```
chi2 = 0.0   # R's implementation
p-value = 1.0  # R's implementation
```

Proposed implementation of R's correction:

`chi2_contingency(X, correction=True)` gives us:
```
chi2 = 0.0024999752722525003   # proposed implementation
p-value =  0.9601225853761536  # proposed implementation
```
So,  R's correction gives us edge-case p-value (p-val=1.0), that may be treated as something suspicious (don't know the best word for this). But proposed implementation just slightly increase p-value toward 1, that looks more suitable for the term "correction" (at least, for my opinion).  

What do you think about proposed implementation: ` yates = 0.5 * np.minimum(1.0, np.abs(expected - observed))`?

Reply:
As Ralf pointed out, testing agains R is not the problem. But just talking about re-implementing some code which is under GPL is dangerous. And there is now a trace about such things here. IMHO you should close this PR and open a new one without mentioning this. Instead refer to the relevant literature and come up with the most natural way to translate Yates' correction. During the review we can then propose improvement only based on what you propose and numerical considerations.

It's also why I would prefer not to comment any code here as to avoid any conflict. Thanks for having a look at this.